superset-frontend/src/utils/getDatasourceUid.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Dataset } from '@superset-ui/chart-controls';
+
+export const getDatasourceUid = (datasource: Dataset & { uid?: string }) =>
+  datasource.uid ?? `${datasource.id}__${datasource.type}`;
```

Comment:
Also, does that mean that this is not required anymore?

<img width="576" alt="image" src="https://user-images.githubusercontent.com/15073128/175037911-68fbb831-62b2-4e5f-a2fb-46d5ed9c2281.png">

Reply:
Also side question, but is the `datasource.uid` the same string construct of id and type or a UUID string?

Reply:
`superset/connectors/base/models.py` composes uid as `${id}__${type}`. I created `getDatasourceUid` function because `uid` wasn't in `Dataset` interface in frontend. It's probably a mistake and uid is always present, but I added composing it manually as a failsafe

<img width="467" alt="image" src="https://user-images.githubusercontent.com/15073128/175285835-e7052c81-fdc9-4573-a18c-0af8daf6940c.png">

Reply:
ok, perfect!